juspay/hyperswitch
# customers: Added ephemeral key authentication for customer retrieve

## Type of Change


## Description

Added a function to authenticate customer retrieve through Ephemeral Key.

### Additional Changes





## Motivation and Context

Ephemeral key provides temporary access to customer database for 1hour and to a particular customer_id.
This allows SDK to access the customer table without leaking API-Key temporarily.

## How did you test it?

- Create an ephemeral Key with some `customer_id`.
- And try to retrieve the `customer_id` from the key successfully fetched.
- Try to fetch any other `customer_id` from the epkey failed like expected.

## Checklist


- [x] I formatted the code `cargo +nightly fmt`
